### ISSUE TITLE
`no-immutable-reactive-statements` false positive on mutable `const`s that are only mutated by a binding

### ISSUE THREAD

**Issue by DetachHead** [sveltejs/eslint-plugin-svelte#584]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.24.0

### What version of `eslint-plugin-svelte` are you using?

2.33.1

### What did you do?

```svelte
<script>
  const values = ['a', 'b']

  $: first = values[0] // error: no-immutable-reactive-statements
</script>

{first}

{#each values as value}
 <input bind:value>
{/each}
```

### What did you expect to happen?

no error, because the value is being mutated

### What actually happened?

`no-immutable-reactive-statements` error on the reactive statement

### Link to **GitHub Repo** with Minimal Reproducible Example

https://sveltejs.github.io/eslint-plugin-svelte/playground/#eJw9zE0OgyAQBeCrTGgTNhqaLon2IuoCcZqS4GBgdGO4e9H+7F7mffN2YcOEQosm2egWfvQEYAMlhs34FRO00EkjK5CjHHo66quGp4tFtF/T3QZQCjDGEHtq1H+qp/2U+YwXNPb1mzXpk0oFjaNlZRgdTfo8ltddHTqLSsTVYxJ6F2lDz6go1G6eVzajxzoWxW7DOrFhnJG40HvObzyKTGI=

### Additional comments

related: #580
